source/set-required-deep.d.ts

```diff
@@ -0,0 +1,45 @@
+import type {OmitDeep} from './omit-deep';
+import type {Paths} from './paths';
+import type {PickDeep} from './pick-deep';
+import type {RequiredDeep} from './required-deep';
+import type {SimplifyDeep} from './simplify-deep';
+
+/**
+Create a type that makes the given keys required. You can specifiy deeply nested key paths. The remaining keys are kept as is.
+
+Use-case: You want to define as required only a some nested keys inside a model.
+
+@example
+```
+import type {SetRequiredDeep} from 'type-fest';
+
+type Foo = {
+	a?: number;
+	b?: string;
+	array?: {
+		c?: number
+	}[]
+}
+
+type SomeRequiredDeep = SetRequiredDeep<Foo, 'a' | 'array.c'>;
```

Comment:
```suggestion
type SomeRequiredDeep = SetRequiredDeep<Foo, 'a' | 'array.${number}.c'>;
```

I kinda forgot how it's should be. You can add the example in test to ensure it is worked.

Reply:
> I kinda forgot how it's should be. You can add the example in test to ensure it is worked.

This doesn't seem to be resolved @hugomartinet